pkg/remediator/reconcile/worker.go

```diff
@@ -35,15 +35,22 @@ import (
 // Worker pulls objects from a work queue and passes them to its reconciler for
 // remediation.
 type Worker struct {
-	objectQueue queue.Interface
-	reconciler  reconcilerInterface
+	objectQueue          queue.Interface
+	reconciler           reconcilerInterface
+	addFightErrorFunc    func(id core.ID, e status.Error)
```

Comment:
Not a big fan of injecting functions like this. It exports business logic and makes it hard to find/debug.

Is there any way to avoid it by wrapping the Worker or injecting a FightHandler "strategy pattern" struct?

Reply:
Changed to use a FightHandler to update the fight errors.